constraint-solver/src/solver.rs

```diff
@@ -203,13 +229,58 @@ impl<T: FieldElement, V: Ord + Clone + Hash + Eq + Display + Debug> Solver<T, V>
             false
         }
     }
+
+    /// Given a set of variable assignments, checks if they satisfy all the constraints.
+    /// Note that this might return false positives, because it does not propagate any values.
+    fn check_assignments(&self, assignments: &BTreeMap<V, T>) -> bool {
+        let constraints = self
+            .constraint_system
+            .get_constraints(assignments.keys().cloned());
+
+        for constraint in constraints {
+            match constraint {
+                ConstraintSystemItem::AlgebraicConstraint(index) => {
+                    let mut identity =
+                        self.constraint_system.algebraic_constraints()[index].clone();
+                    for (variable, value) in assignments.iter() {
+                        identity
+                            .substitute_by_known(variable, &SymbolicExpression::Concrete(*value));
+                    }
+                    if identity.solve(&self.range_constraints).is_err() {
+                        return false;
+                    }
+                }
+                ConstraintSystemItem::BusInteraction(index) => {
+                    let mut bus_interaction =
+                        self.constraint_system.bus_interactions()[index].clone();
+                    for (variable, value) in assignments.iter() {
+                        bus_interaction.iter_mut().for_each(|expr| {
+                            expr.substitute_by_known(
+                                variable,
+                                &SymbolicExpression::Concrete(*value),
+                            )
+                        })
+                    }
+                    if bus_interaction
+                        .solve(&*self.bus_interaction_handler, &self.range_constraints)
+                        .is_err()
+                    {
+                        return false;
+                    }
+                }
+            }
```

Comment:
`.is_err() { return false }` could move here

Reply:
Not quite, because the error type is different. But I refactored the function, now using `any()`.